bears/hypertext/PHPCodeSnifferBear.py

```diff
@@ -0,0 +1,37 @@
+from coalib.bearlib.abstractions.Linter import linter
+from coalib.bearlib.spacing.SpacingHelper import SpacingHelper
+from coalib.bears.requirements.DistributionRequirement import (
+    DistributionRequirement)
+
+
+@linter(executable='phpcs',
+        output_format='regex',
+        output_regex=r'(?P<filename>.+):(?P<line>\d+):(?P<column>\d+): '
+                      '(\w+) - (?P<message>.+)')
+class PHPCodeSnifferBear:
+    """
+    Tokenises and "sniffs" PHP, JavaScript and CSS files to detect violations
+    of a defined coding standard.
+
+    See <http://pear.php.net/manual/en/package.php.php-codesniffer.php> for
+    more information.
+    """
+
+    LANGUAGES = {"PHP", "JavaScript", "CSS"}
+    REQUIREMENTS = {DistributionRequirement(apt_get='php-codesniffer')}
+    AUTHORS = {'The coala developers'}
+    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
+    LICENSE = 'AGPL-3.0'
+    CAN_DETECT = {'Formatting', 'Syntax', 'Documentation',
+                  'Code Simplification'}
+
+    @staticmethod
+    def create_arguments(filename, file, config_file,
+                         indent_size: int=SpacingHelper.DEFAULT_TAB_WIDTH,
+                         use_spaces: bool=True):
+        """
+        :param indent_size:  Number of spaces per indentation level.
+        :param use_spaces:   True if spaces are to be used, else tabs.
+        """
+        options = '--tab-width=' + indent_size if not use_spaces else ''
```

Comment:
ouh, this looks dangeous: https://github.com/squizlabs/PHP_CodeSniffer/wiki/Configuration-Options#setting-the-default-tab-width apparently it modifies the file!?